lua/caching_handlers.lua

```diff
@@ -171,6 +171,12 @@ function caching_handlers._parse_request(incoming_zipkin_headers)
         ngx.req.clear_header("accept-encoding")
         request_headers['accept-encoding'] = nil
     end
+    -- Let's also remove the 'application/msgpack' Accept header so that it's easier to
+    -- parse the response.
+    if ngx.re.match(ngx.req.get_headers()['accept'], 'application/msgpack') then
```

Comment:
This means that the whole "Accept" header is going to get dropped if "application/msgpack" is somewhere in it. If that's fine, shipit!

If not let's add a test for "Accept" headers with multiple values (say `Accept: application/msgpack, text/html, application/xhtml+xml` -- see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept)

Reply:
Technically we only accept json, so any other Accept header is invalid. But we never assert that anywhere. Also right now we're dropping both gzip and Accept also for non-bulk endpoints which would handle both without problems.

I'll try to move this code so that it only affects bulk endpoints and assert that everything is JSON.

Reply:
Right now this header is 100% of the time `Accept: */*` as casper sees it. But I agree we probably don't want to drop the Accept header entirely.